fe/fe-core/src/main/java/com/starrocks/external/hive/events/MetastoreEventFactory.java

```diff
@@ -31,68 +30,60 @@ public class MetastoreEventFactory implements EventFactory {
             case ALTER_PARTITION:
-                return AlterPartitionEvent.getEvents(event, metaCache);
+                return AlterPartitionEvent.getEvents(event, metaCache, catalogName);
             case DROP_PARTITION:
-                return DropPartitionEvent.getEvents(event, metaCache, table.getPartitionColumns());
+                return DropPartitionEvent.getEvents(event, metaCache, table.getPartitionColumns(), catalogName);
             case INSERT:
-                return InsertEvent.getEvents(event, metaCache);
+                return InsertEvent.getEvents(event, metaCache, catalogName);
             default:
                 // ignore all the unknown events by creating a IgnoredEvent
-                return Lists.newArrayList(new IgnoredEvent(event, metaCache));
+                return Lists.newArrayList(new IgnoredEvent(event, metaCache, catalogName));
         }
     }
 
-    List<MetastoreEvent> getFilteredEvents(List<NotificationEvent> events, String resourceName) {
+    List<MetastoreEvent> getFilteredEvents(List<NotificationEvent> events,
+                                           CacheUpdateProcessor metaCache, String catalogName) {
         List<MetastoreEvent> metastoreEvents = Lists.newArrayList();
-        CacheUpdateProcessor metaCache = null;
-        // TODO(stephen): refactor auto sync hive metatadata
-        // metaCache = GlobalStateMgr.getCurrentState().getHiveRepository().getMetaCache(resourceName);
 
         if (metaCache == null) {
-            LOG.error("Meta cache is null on resource [{}]", resourceName);
+            LOG.error("Meta cache is null on catalog [{}]", catalogName);
             return metastoreEvents;
         }
 
         // Currently, the hive external table needs to be manually created in StarRocks to map with the hms table.
         // Therefore, it's necessary to filter the events pulled this time from the hms instance,
         // and the events of the tables that don't register in the fe MetastoreEventsProcessor need to be filtered out.
         for (NotificationEvent event : events) {
-            HiveTable table = null;
-            if (IcebergTable.isInternalCatalog(resourceName)) {
-                table = GlobalStateMgr.getCurrentState().getMetastoreEventsProcessor()
-                        .getHiveTable(resourceName, event.getDbName(), event.getTableName());
-            } else {
-                // TODO(stephen): refactor auto sync hive metatadata
-                // table = (HiveTable) metaCache.getTableFromCache(HiveTableName.of(event.getDbName(), event.getTableName()));
-            }
-
-            if (table == null) {
-                LOG.warn("Table is null on resource [{}], table [{}.{}]. Skipping notification event {}",
-                        resourceName, event.getDbName(), event.getTableName(), event);
+            if (!metaCache.existIncache(MetastoreEventType.ALTER_TABLE,
```

Comment:
The conditions are not strict, the events of the external table will be filtered out